src/arena.rs

```diff
@@ -119,36 +125,36 @@ mod tests {
 
     #[test]
     fn append_non_unique() {
-        let mut arena: Arena<f64> = Arena::new();
-        let t1 = arena.append(0.0);
-        let t2 = arena.append(0.0);
+        let mut arena: Arena<u8> = Arena::new();
+        let t1 = arena.append(0);
+        let t2 = arena.append(0);
         assert!(t1 != t2);
         assert!(arena[t1] == arena[t2]);
     }
 
     #[test]
     fn append_unique() {
-        let mut arena: Arena<f64> = Arena::new();
-        let t1 = arena.append(std::f64::NAN);
-        let t2 = arena.append(std::f64::NAN);
+        let mut arena: Arena<u8> = Arena::new();
+        let t1 = arena.append(0);
+        let t2 = arena.append(1);
         assert!(t1 != t2);
         assert!(arena[t1] != arena[t2]);
     }
 
     #[test]
     fn fetch_or_append_non_unique() {
-        let mut arena: Arena<f64> = Arena::new();
-        let t1 = arena.fetch_or_append(0.0);
-        let t2 = arena.fetch_or_append(0.0);
+        let mut arena: Arena<u8> = Arena::new();
+        let t1 = arena.fetch_or_append(0);
+        let t2 = arena.fetch_or_append(0);
         assert!(t1 == t2);
         assert!(arena[t1] == arena[t2])
     }
 
     #[test]
     fn fetch_or_append_unique() {
-        let mut arena: Arena<f64> = Arena::new();
-        let t1 = arena.fetch_or_append(std::f64::NAN);
-        let t2 = arena.fetch_or_append(std::f64::NAN);
+        let mut arena: Arena<u8> = Arena::new();
```

Comment:
what's the motivation for these test changes?

Reply:
Float equality checxking is a thorny topic and it didn't seem to benefit these tests (I'll happily be corrected on that point though). The reason I noticed was because clippy was angry (although it can just be silenced which would be perfectly valid here).

Reply:
makes sense 👍